metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java

```diff
@@ -180,15 +183,45 @@ private void updateFineGrainedEdgesAndRelationships(
     }
   }
 
+  private Urn generateSchemaFieldUrn(@Nonnull String resourceUrn, @Nonnull String fieldPath) {
+    // we rely on schemaField fieldPaths to be encoded since we do that with fineGrainedLineage on the ingestion side
+    String encodedFieldPath = fieldPath.replaceAll("\\(", "%28").replaceAll("\\)", "%29").replaceAll(",", "%2C");
+    String urnString = String.format("urn:li:schemaField:(%s,%s)", resourceUrn, encodedFieldPath);
+    return UrnUtils.getUrn(urnString);
+  }
+
+  private void updateInputFieldEdgesAndRelationships(
+      Urn urn,
+      RecordTemplate aspect,
+      List<Edge> edgesToAdd,
+      HashMap<Urn, Set<String>> urnToRelationshipTypesBeingAdded
+  ) {
+    InputFields inputFields = new InputFields(aspect.data());
+    if (inputFields.hasFields()) {
+      for (InputField field : inputFields.getFields()) {
+        if (field.hasSchemaFieldUrn() && field.hasSchemaField() && field.getSchemaField().hasFieldPath()) {
```

Comment:
Doesn't this already have a schemaFieldUrn in this case? Why cannot we use this URN? Is it _not_ encoded?

Reply:
this is part of the confusing modeling thing here - `schemaFieldUrn` is actually the upstream urn for this inputField and it could be null. So the `schemaFieldUrn` and the `schemaField` properties here are going to be referencing two different schemaFields

Reply:
and the `schemaFieldUrn` is encoded

Reply:
got it ty